src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,401 @@
+                    {
+                        if (!IsSuppressedByAttribute(parsedAttribute, context.ContainingSymbol))
+                        {
+                            builder.Add(parsedAttribute);
+                        }
+                    }
+                }
+            }
+
+            if (builder.Count > 0)
+            {
+                platformSpecificOperations.Add(operation, builder.ToImmutable());
+            }
+        }
+
+        private static bool IsWithinConditionalOperation(IOperation pOperation) =>
+            pOperation.ConstantValue.HasValue &&
+            pOperation.Parent is IBinaryOperation bo &&
+            (bo.OperatorKind == BinaryOperatorKind.Equals ||
+            bo.OperatorKind == BinaryOperatorKind.NotEquals ||
+            bo.OperatorKind == BinaryOperatorKind.GreaterThan ||
+            bo.OperatorKind == BinaryOperatorKind.LessThan ||
+            bo.OperatorKind == BinaryOperatorKind.GreaterThanOrEqual ||
+            bo.OperatorKind == BinaryOperatorKind.LessThanOrEqual);
+
+        private static bool IsOSPlatformsEqual(string left, string right)
+        {
+            if (left.Equals(right, StringComparison.OrdinalIgnoreCase))
+            {
+                return true;
+            }
+            else
+            {
+                var normalizedLeft = NormalizeOSName(left);
+                var normalizedRight = NormalizeOSName(right);
+                return string.Equals(normalizedLeft, normalizedRight, StringComparison.OrdinalIgnoreCase);
+            }
+        }
+
+        private static string NormalizeOSName(string name) => name.Equals(OSX, StringComparison.OrdinalIgnoreCase)
+            ? macOS
+            : name;
+
+        private static DiagnosticDescriptor SelectRule(PlatformAttributeType attributeType)
+            => attributeType switch
+            {
+                PlatformAttributeType.MinimumOSPlatformAttribute => MinimumOsRule,
+                PlatformAttributeType.ObsoletedInOSPlatformAttribute => ObsoleteRule,
+                _ => RemovedRule,
+            };
+
+        private static ImmutableArray<AttributeData> FindAllPlatformAttributesApplied(ImmutableArray<AttributeData> immediateAttributes, INamedTypeSymbol parent)
+        {
+            using var builder = ArrayBuilder<AttributeData>.GetInstance();
+            foreach (AttributeData attribute in immediateAttributes)
+            {
+                if (s_osPlatformAttributes.Contains(attribute.AttributeClass.Name))
+                {
+                    builder.Add(attribute);
+                }
+            }
+
+            while (parent != null)
+            {
+                var current = parent.GetAttributes();
+                foreach (var attribute in current)
+                {
+                    if (s_osPlatformAttributes.Contains(attribute.AttributeClass.Name) &&
+                        TargetPlatformAttribute != attribute.AttributeClass.Name)
```

Comment:
Can we add a comment as to why are we skipping the `TargetPlatformAttribute` for containing symbols but not for the primary symbol?

Reply:
Leftover from old logic, where we were checking by parent Attribute type, will be removed

Reply:
Great, in that case I think you can move the for loop that processes attributes here and just above in this same method into a local function.